file.go

```diff
@@ -1,37 +1,121 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
 
 type file struct {
 	// TODO: define variables if needed
+	JSONFormat map[string]interface{}
+	CSVFormat  struct {
+		Data [][]string
+		//if "headers==true", data[0] will contain headers
+		HasHeaders bool
+	}
 }
 
+func (f *file) initialize() {
+	f.JSONFormat = nil
+	f.CSVFormat.Data = nil
+	f.CSVFormat.HasHeaders = false
+}
 func ParseJsonFile(path string) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	jsonFile, err := os.Open(path)
+	var myFile file
+	myFile.initialize()
+	if err != nil {
+		return myFile, err
+	}
+	defer jsonFile.Close()
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err == nil {
+		json.Unmarshal([]byte(byteValue), &myFile.JSONFormat)
+	}
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
 	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var myFile file
+	myFile.initialize()
+	byteValue, err := ioutil.ReadAll(f)
+	if err == nil {
+		json.Unmarshal([]byte(byteValue), &myFile.JSONFormat)
+	}
+	return myFile, err
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	csvFile, err := os.Open((path))
+	var myFile file
+	myFile.initialize()
+	if err != nil {
+		return myFile, err
+	}
+	defer csvFile.Close()
+	csvReader := csv.NewReader(csvFile)
```

Comment:
And an empty line between each code block to make your code look cleaner.

Reply:
Sure, I will keep that in mind from now on.